props/hev.h

```diff
@@ -0,0 +1,440 @@
+      // Make armor compromised sound play before setting armor to 0
+      SaberBase::DoEffect(EFFECT_USER2, 0.0, 0);
+      PVLOG_NORMAL << "Armor Compromised!\n";
+      armor_ = 0;
+    } else {
+      health_ -= damage;
+    }
+
+    // Enforce minimum values
+    if (health_ < 0) health_ = 0;
+    if (armor_ < 0) armor_ = 0;
+
+    // Damage
+    if (!quiet) SaberBase::DoEffect(EFFECT_STUN, 0.0, damage);
+    
+    // Death Sound
+    if (health_ == 0) {
+      SaberBase::DoEffect(EFFECT_EMPTY, 0.0, 100);
+      return;
+    }
+    
+    // Health Alert - only plays when health enters a new multiple of 10
+    int new_tens = health_ / 10;
+    if (tens != new_tens) {
+      SaberBase::DoEffect(EFFECT_USER1, 0.0, 0);
+    }
+
+    // Print health and armor
+    PVLOG_NORMAL << "Health: " << health_ << " / ";
+    PVLOG_NORMAL << "Armor: " << armor_ << "\n";
+  }
+
+
+  // Armor Readout /////////////////////////////////////////////////////////////
+  void armor_readout() {
+    PVLOG_NORMAL << "Current Armor: " << armor_ << "\n"; // Debug logging
+    SFX_armor.SelectFloat(armor_ / 100.0);
+    hybrid_font.PlayCommon(&SFX_armor);
+  }
+
+
+  // Clashes ///////////////////////////////////////////////////////////////////
+  uint32_t last_clash_time_ = 0;
+  const uint32_t CLASH_DEBOUNCE_MS = 100; // Adjust debounce time as needed
+
+  void Clash(bool stab, float strength) override {
+    // Ignore Clash if within debounce period
+    uint32_t current_time = millis();
+    if (current_time - last_clash_time_ < CLASH_DEBOUNCE_MS) {
+      return;
+    }
+    last_clash_time_ = current_time;
+
+    // Stop Clashes if health is 0
+    if (health_ == 0) {
+      return;
+    }
+
+     // Strength multiplier
+    int damage = (int)(strength * 4);
+
+    // Play Clash sounds based on strength
+    float v = (strength - GetCurrentClashThreshold()) / 3;
+    SFX_clash.SelectFloat(v);
+    SFX_clsh.SelectFloat(v);
+    SFX_stab.SelectFloat(v);
+
+    // Cap damage at 50
+    if (damage > 50) {
+      damage = 50;
+    }
+
+    // Play Armor Alarm if damage is 30 or more
+    if (damage >= 30) {
+      hybrid_font.PlayPolyphonic(&SFX_armor_alarm);
+    }
+
+    // Apply Physical Damage and print to log
+    DoDamage(damage, true);
+    PVLOG_NORMAL << "Physical Damage: -" << damage << "\n";
+
+    // No stabs!
+    PropBase::Clash(false, strength);
+  }
+
+
+  // Swings do nothing! ////////////////////////////////////////////////////
+  void DoMotion(const Vec3& motion, bool clear) override {
+    PropBase::DoMotion(Vec3(0), clear);
+  }
+
+
+  // Random Hazards ////////////////////////////////////////////////////////////
+  uint32_t last_random_draw_ = millis();
+  uint32_t post_death_cooldown_ = 0;
+
+  // Check and activation
+  void CheckRandomEvent() {
+    // Don't allow Hazards if dead or during post-death cooldown.
+    // This prevents Hazards immediately after healing.
+    if (health_ == 0 || (post_death_cooldown_ &&
```

Comment:
millis will eventually return zero, it's better to have a separate bool for if the cooldown is "active" instead of re-using the timeout value.